### ISSUE TITLE
3.7.0: Cannot read property 'callee' of null

### ISSUE THREAD

**Issue by psbanka** [yannickcr/eslint-plugin-react#286]
Just upgraded to the latest eslint-plugin-react and I'm getting this error when trying to lint. Version 3.6.3 did not exhibit this issue. This issue is happening with latest `eslint` using node `v0.10.29` and node `v4.2.1`

**Comment by yannickcr** [comment]
Can you give me more details about the error ? Like the source code that trigger this error or at least the error stack trace. Thanks.

**Comment by themaxx** [comment]
Might be blowing up here: https://github.com/yannickcr/eslint-plugin-react/blob/master/lib/util/Components.js#L113

**Comment by jerrysu** [comment]
Same issue:

```
/Users/jerrysu/Remind/r101-frontend/node_modules/eslint-plugin-react/lib/util/Components.js:113
      return sourceCode.getText(node.parent.callee) === 'React.createClass';
                                           ^
TypeError: Cannot read property 'callee' of null
    at Object.context.react.isES5Component (/Users/jerrysu/Remind/r101-frontend/node_modules/eslint-plugin-react/lib/util/Components.js:113:44)
    at Object.context.react.getParentES5Component (/Users/jerrysu/Remind/r101-frontend/node_modules/eslint-plugin-react/lib/util/Components.js:183:35)
    at isPropTypesUsage (/Users/jerrysu/Remind/r101-frontend/node_modules/eslint-plugin-react/lib/rules/prop-types.js:31:63)
    at EventEmitter.MemberExpression (/Users/jerrysu/Remind/r101-frontend/node_modules/eslint-plugin-react/lib/rules/prop-types.js:561:11)
    at EventEmitter.emit (events.js:129:20)
    at NodeEventGenerator.enterNode (/Users/jerrysu/Remind/r101-frontend/node_modules/eslint/lib/util/node-event-generator.js:42:22)
    at CommentEventGenerator.enterNode (/Users/jerrysu/Remind/r101-frontend/node_modules/eslint/lib/util/comment-event-generator.js:98:23)
    at Controller.controller.traverse.enter (/Users/jerrysu/Remind/r101-frontend/node_modules/eslint/lib/eslint.js:782:36)
    at Controller.__execute (/Users/jerrysu/Remind/r101-frontend/node_modules/eslint/node_modules/estraverse/estraverse.js:397:31)
    at Controller.traverse (/Users/jerrysu/Remind/r101-frontend/node_modules/eslint/node_modules/estraverse/estraverse.js:495:28)
```

**Comment by oliversong** [comment]
+1

**Comment by yannickcr** [comment]
Thanks for the details, this should be fixed in next release.

**Comment by Zenwolf** [comment]
I am getting the same error after installing version `3.7.0`:
```
/Users/mjaquish/git/shared-ui/shared-ui/node_modules/eslint-plugin-react/lib/util/Components.js:113
      return sourceCode.getText(node.parent.callee) === 'React.createClass';
                                           ^
TypeError: Cannot read property 'callee' of null
    at Object.context.react.isES5Component (/Users/mjaquish/git/shared-ui/shared-ui/node_modules/eslint-plugin-react/lib/util/Components.js:113:44)
    at Object.context.react.getParentES5Component (/Users/mjaquish/git/shared-ui/shared-ui/node_modules/eslint-plugin-react/lib/util/Components.js:183:35)
    at Object.context.react.getParentComponent (/Users/mjaquish/git/shared-ui/shared-ui/node_modules/eslint-plugin-react/lib/util/Components.js:170:23)
    at Object.detectionInstructions.FunctionDeclaration (/Users/mjaquish/git/shared-ui/shared-ui/node_modules/eslint-plugin-react/lib/util/Components.js:337:28)
    at EventEmitter.updatedRuleInstructions.(anonymous function) (/Users/mjaquish/git/shared-ui/shared-ui/node_modules/eslint-plugin-react/lib/util/Components.js:375:41)
    at EventEmitter.emit (events.js:117:20)
    at NodeEventGenerator.enterNode (/Users/mjaquish/git/shared-ui/shared-ui/node_modules/eslint/lib/util/node-event-generator.js:42:22)
    at CommentEventGenerator.enterNode (/Users/mjaquish/git/shared-ui/shared-ui/node_modules/eslint/lib/util/comment-event-generator.js:98:23)
    at Controller.controller.traverse.enter (/Users/mjaquish/git/shared-ui/shared-ui/node_modules/eslint/lib/eslint.js:782:36)
    at Controller.__execute (/Users/mjaquish/git/shared-ui/shared-ui/node_modules/eslint/node_modules/estraverse/estraverse.js:397:31)
    at Controller.traverse (/Users/mjaquish/git/shared-ui/shared-ui/node_modules/eslint/node_modules/estraverse/estraverse.js:495:28)
```

**Comment by yannickcr** [comment]
@Zenwolf this issue was fixed in `v3.7.1`, upgrade and it should be ok :)

**Comment by Zenwolf** [comment]
@yannickcr Ah, great! My mirror hadn't picked up that version yet :) Thanks!
